test-framework/core/src/main/java/org/keycloak/testframework/realm/UserConfigBuilder.java

```diff
@@ -87,6 +91,31 @@ public UserConfigBuilder attribute(String key, String... value) {
         return this;
     }
 
+    public UserConfigBuilder federatedLink(String identityProvider, String federatedUserId) {
+        if (rep.getFederatedIdentities() == null) {
+            rep.setFederatedIdentities(new LinkedList<>());
+        }
+        FederatedIdentityRepresentation federatedIdentity = new FederatedIdentityRepresentation();
+        federatedIdentity.setUserId(federatedUserId);
+        federatedIdentity.setUserName(rep.getUsername());
```

Comment:
`FederatedIdentityRepresentation#userName` should be the username from the external IdP, and not the local user, so you probably need to pass `String federatedUsername`

```suggestion
        federatedIdentity.setUserName(rep.getUsername());
```